nexus/reconfigurator/planning/src/planner.rs

```diff
@@ -642,7 +642,7 @@ mod test {
         let zones = sled_changes.zones_added().collect::<Vec<_>>();
         assert_eq!(zones.len(), policy.target_nexus_zone_count - 1);
         for zone in &zones {
-            let OmicronZoneType::Nexus { .. } = zone.zone_type else {
+            let OmicronZoneType::Nexus { .. } = zone.config.zone_type else {
```

Comment:
Only because the diffs made this obvious: should this be

```suggestion
            if !zone.config.zone_type.is_nexus() {
```

instead?

Reply:
Done, and also added `is_crucible` and changed another use of this pattern elsewhere.